xcp_d/interfaces/qc_plot.py

```diff
@@ -22,7 +23,175 @@
+        exists=False,
+        mandatory=True,
+        desc="Framewise displacement threshold."
+    )
+
+
+class _CensoringPlotOutputSpec(TraitedSpec):
+    out_file = File(exists=True, mandatory=True, desc="Censoring plot.")
+
+
+class CensoringPlot(SimpleInterface):
+    """Generate a censoring figure.
+
+    This is a line plot showing both the raw and filtered framewise displacement time series,
+    with vertical lines/bands indicating volumes removed by the post-processing workflow.
+    """
+
+    input_spec = _CensoringPlotInputSpec
+    output_spec = _CensoringPlotOutputSpec
+
+    def _run_interface(self, runtime):
+        # Load confound matrix and load motion with motion filtering
+        confound_matrix = load_confound(datafile=self.inputs.bold_file)[0]
+        preproc_motion_df = load_motion(
+            confound_matrix.copy(),
+            TR=self.inputs.TR,
+            motion_filter_type=None,
+        )
+        preproc_fd_timeseries = compute_fd(
+            confound=preproc_motion_df,
+            head_radius=self.inputs.head_radius,
+        )
+
+        fig, ax = plt.subplots(figsize=(16, 8))
+
+        time_array = np.arange(0, self.inputs.TR * preproc_fd_timeseries.size, self.inputs.TR)
+
+        ax.plot(
+            time_array,
+            preproc_fd_timeseries,
+            label="Raw Framewise Displacement",
+            color="blue",
+        )
+        ax.axhline(self.inputs.fd_thresh, label="Outlier Threshold", color="gray", alpha=0.5)
+
+        if self.inputs.dummytime:
+            initial_volumes_to_drop = int(np.ceil(self.inputs.dummytime / self.inputs.TR))
+            ax.axvspan(
+                0,
+                initial_volumes_to_drop,
+                label="Dummy Volumes",
+                alpha=0.5,
+                color="orange",
+            )
+        else:
+            initial_volumes_to_drop = 0
+
+        # Compute filtered framewise displacement to plot censoring
+        if self.inputs.motion_filter_type:
+            filtered_motion_df = load_motion(
+                confound_matrix.copy(),
+                TR=self.inputs.TR,
+                motion_filter_type=self.inputs.motion_filter_type,
+                motion_filter_order=self.inputs.motion_filter_order,
+                band_stop_min=self.inputs.band_stop_min,
+                band_stop_max=self.inputs.band_stop_max,
+            )
+            filtered_fd_timeseries = compute_fd(
+                confound=filtered_motion_df,
+                head_radius=self.inputs.head_radius,
+            )
+
+            ax.plot(
+                time_array,
+                filtered_fd_timeseries,
+                label="Filtered Framewise Displacement",
+                color="red",
+            )
+        else:
+            filtered_fd_timeseries = preproc_fd_timeseries.copy()
+
+        # NOTE: TS- Probably should replace with the actual tmask file.
+        tmask = filtered_fd_timeseries >= self.inputs.fd_thresh
```

Comment:
Yes, would agree - not sure if we wanna do that in this PR/ low priority?

Reply:
It's pretty low priority given how simple the censoring is. I'd rather tackle it in a future PR.